jetty-security/src/main/java/org/eclipse/jetty/security/DefaultAuthenticatorFactory.java

```diff
@@ -64,7 +66,17 @@ else if (Constraint.__SPNEGO_AUTH.equalsIgnoreCase(auth))
         else if (Constraint.__NEGOTIATE_AUTH.equalsIgnoreCase(auth)) // see Bug #377076
             authenticator = new ConfigurableSpnegoAuthenticator(Constraint.__NEGOTIATE_AUTH);
         if (Constraint.__CERT_AUTH.equalsIgnoreCase(auth) || Constraint.__CERT_AUTH2.equalsIgnoreCase(auth))
-            authenticator = new ClientCertAuthenticator();
+        {
+            Collection<SslContextFactory> sslContextFactories = server.getBeans(SslContextFactory.class);
+            if (sslContextFactories.size() != 1)
+            {
+                authenticator = new ClientCertAuthenticator();
```

Comment:
maybe we should have a warning here?

Reply:
I think at least a debug saying in both cases how the authenticator is constructed.   If there are >2 SslContextFactories, then a warning is a appropriate as we are picking neither and they really should configure a specific authenticator.